releasenotes/notes/perturbation-module-b819dfac35c60551.yaml

```diff
@@ -0,0 +1,24 @@
+---
+prelude: >
+    Adds perturbation theory sub module for computing and utilizing multi-variable power series
+    expansions commonly used when studying quantum dynamics, e.g. Dyson series and Magnus expansion,
+    and the generator of the Schrieffer-Wolff transformation.
+features:
+  - |
+    Adds :class:`~qiskit_dynamics.perturbation.ArrayPolynomial` for representing
+    multi-variable array-valued polynomials.
+  - |
+    Adds function :func:`~qiskit_dynamics.perturbation.solve_lmde_perturbation` for computing
+    Dyson series and Magnus expansion terms for solutions to linear matrix
+    differential equations.
+  - |
+    Adds data container class :class:`qiskit_dynamics.perturbation.PerturbationResults`
+    for storing and accessing the results of
+    :func:`~qiskit_dynamics.perturbation.solve_lmde_perturbation`.
+  - |
+    Adds class :class:`~qiskit_dynamics.perturbation.PerturbativeSolver`, which implements
+    specialized fixed-step solvers for linear matrix differential equations,
+    based on the Dyson series and Magnus expansion.
+other:
```

Comment:
This probably is an upgrade note

Reply:
Do you mean here that everything under "features" should be under "upgrade"?

Reply:
Changed to upgrade - after talking to @mtreinish  I've realized I misunderstood the typical qiskit usage of `upgrade` section.